test/e2e/network/util.go

```diff
@@ -53,6 +53,23 @@ func GetHTTPContent(host string, port int, timeout time.Duration, url string) by
 	return body
 }
 
+// GetHTTPContentFromTestContainer returns the content of the given url by HTTP via a test container.
+func GetHTTPContentFromTestContainer(config *e2enetwork.NetworkingTestConfig, host string, port int, timeout time.Duration, dialCmd string) string {
+	var body string
+	pollFn := func() (bool, error) {
+		resp := config.GetResponseFromTestContainer("http", dialCmd, host, port)
+		if resp == nil || len(resp.Errors) > 0 || len(resp.Responses) == 0 {
+			return false, nil
+		}
+		body = resp.Responses[0]
+		return true, nil
+	}
+	if pollErr := wait.PollImmediate(framework.Poll, timeout, pollFn); pollErr != nil {
+		framework.Failf("Could not reach HTTP service through %v:%v/%v after %v: %v", host, port, dialCmd, timeout, pollErr)
```

Comment:
I'm not a fan of failing from those kind of methods. Instad, I would prefer returning an error here and failing from the place where it is called (which is the test itself).